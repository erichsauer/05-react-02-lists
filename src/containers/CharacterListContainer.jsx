import React, { Component } from 'react';
import Character from '../components/Character';
import CharacterList from '../components/CharacterList';
import { getCharacters } from '../services/apiCall';

export default class CharacterListContainer extends Component {
  state = {
    characters: [],
    loading: true,
  };

  async componentDidMount() {
    const characters = await getCharacters();
    this.setState({
      characters,
      loading: false,
    });
  }

  render() {
    const { loading, characters } = this.state;

    if (loading) return <div>please hold...</div>;

    return <CharacterList characters={characters} />;
  }
}
