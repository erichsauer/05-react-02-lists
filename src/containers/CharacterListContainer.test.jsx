import React from 'react';
import { render, screen } from '@testing-library/react';
import CharacterListContainer from './CharacterListContainer';
import { MemoryRouter } from 'react-router';

describe('character list container', () => {
  it('renders a list of characters', async () => {
    //   render(
    //     <MemoryRouter>
    //       <CharacterListContainer />
    //     </MemoryRouter>
    //   );
    //   screen.getByText('please hold...');
    //   const ul = await screen.findByRole('list', { name: 'characters' });
    //   expect(ul).toMatchSnapshot();
  });
});
