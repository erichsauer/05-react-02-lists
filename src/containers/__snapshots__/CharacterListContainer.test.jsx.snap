// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`character list container renders a list of characters 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e460"
    >
      <img
        alt="Admiral"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
        style="width: 200px;"
      />
      <p>
        Admiral
      </p>
      <p>
        イッテツ Ittetsu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e461"
    >
      <img
        alt="Agent S"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
        style="width: 200px;"
      />
      <p>
        Agent S
      </p>
      <p>
        2ごう Ni Gō
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e462"
    >
      <img
        alt="Agnes"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
        style="width: 200px;"
      />
      <p>
        Agnes
      </p>
      <p>
        アグネス Agunesu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e463"
    >
      <img
        alt="Al"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
        style="width: 200px;"
      />
      <p>
        Al
      </p>
      <p>
        たもつ Tamotsu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e464"
    >
      <img
        alt="Alfonso"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
        style="width: 200px;"
      />
      <p>
        Alfonso
      </p>
      <p>
        アルベルト Aruberuto
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e465"
    >
      <img
        alt="Alice"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
        style="width: 200px;"
      />
      <p>
        Alice
      </p>
      <p>
        メルボルン Meruborun
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46c"
    >
      <img
        alt="Ankha"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
        style="width: 200px;"
      />
      <p>
        Ankha
      </p>
      <p>
        ナイル Nairu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46a"
    >
      <img
        alt="Angus"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
        style="width: 200px;"
      />
      <p>
        Angus
      </p>
      <p>
        セルバンテス Serubantesu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e466"
    >
      <img
        alt="Alli"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
        style="width: 200px;"
      />
      <p>
        Alli
      </p>
      <p>
        クロコ Kuroko
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46b"
    >
      <img
        alt="Anicotti"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
        style="width: 200px;"
      />
      <p>
        Anicotti
      </p>
      <p>
        ラザニア Razania
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e467"
    >
      <img
        alt="Amelia"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
        style="width: 200px;"
      />
      <p>
        Amelia
      </p>
      <p>
        アンデス Andesu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46d"
    >
      <img
        alt="Annalisa"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
        style="width: 200px;"
      />
      <p>
        Annalisa
      </p>
      <p>
        みやび Miyabi
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46f"
    >
      <img
        alt="Antonio"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
        style="width: 200px;"
      />
      <p>
        Antonio
      </p>
      <p>
        マコト Makoto
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e46e"
    >
      <img
        alt="Annalise"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
        style="width: 200px;"
      />
      <p>
        Annalise
      </p>
      <p>
        シルブプレ Shirubupure
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e468"
    >
      <img
        alt="Anabelle"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
        style="width: 200px;"
      />
      <p>
        Anabelle
      </p>
      <p>
        あるみ Arumi
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e470"
    >
      <img
        alt="Apollo"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
        style="width: 200px;"
      />
      <p>
        Apollo
      </p>
      <p>
        アポロ Aporo
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e469"
    >
      <img
        alt="Anchovy"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
        style="width: 200px;"
      />
      <p>
        Anchovy
      </p>
      <p>
        アンチョビ Anchobi
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e471"
    >
      <img
        alt="Apple"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
        style="width: 200px;"
      />
      <p>
        Apple
      </p>
      <p>
        アップル Appuru
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e472"
    >
      <img
        alt="Astrid"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
        style="width: 200px;"
      />
      <p>
        Astrid
      </p>
      <p>
        キッズ Kizzu
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e473"
    >
      <img
        alt="Audie"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
        style="width: 200px;"
      />
      <p>
        Audie
      </p>
      <p>
        モニカ Monika
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e474"
    >
      <img
        alt="Aurora"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
        style="width: 200px;"
      />
      <p>
        Aurora
      </p>
      <p>
        オーロラ Ōrora
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e476"
    >
      <img
        alt="Avery"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
        style="width: 200px;"
      />
      <p>
        Avery
      </p>
      <p>
        クスケチャ Kusukecha
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e475"
    >
      <img
        alt="Ava"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
        style="width: 200px;"
      />
      <p>
        Ava
      </p>
      <p>
        ドミグラ Domigura
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e47b"
    >
      <img
        alt="Barold"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
        style="width: 200px;"
      />
      <p>
        Barold
      </p>
      <p>
        ニッシー Nisshī
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5f5fb4bbbfd05c2aed82e47a"
    >
      <img
        alt="Bangle"
        src="https://vignette.wikia.nocookie.net/animalcrossing/images/6/68/Bangle_NH.png/revision/latest?cb=20200803120044"
        style="width: 200px;"
      />
      <p>
        Bangle
      </p>
      <p>
        ルーズ Rūzu
      </p>
    </a>
  </li>
</ul>
`;
